presto-main/src/main/java/com/facebook/presto/sql/gen/BytecodeUtils.java

```diff
@@ -194,29 +194,37 @@ else if (type == ConnectorSession.class) {
                 block.append(scope.getVariable("session"));
             }
             else {
-                block.append(arguments.get(realParameterIndex));
+                //block.append(arguments.get(realParameterIndex));
                 ArgumentProperty argumentProperty = function.getArgumentProperty(realParameterIndex);
 
                 if (argumentProperty.getArgumentType() == VALUE_TYPE) {
                     // apply null convention for value type argument
                     switch (argumentProperty.getNullConvention()) {
                         case RETURN_NULL_ON_NULL:
+                            block.append(arguments.get(realParameterIndex));
                             checkArgument(!Primitives.isWrapperType(type), "Non-nullable argument must not be primitive wrapper type");
                             block.append(ifWasNullPopAndGoto(scope, end, unboxedReturnType, Lists.reverse(stackTypes)));
                             break;
                         case USE_NULL_FLAG:
                             if (type == Void.class) {
                                 block.append(boxPrimitiveIfNecessary(scope, type));
                             }
+                            block.append(arguments.get(realParameterIndex));
                             block.append(scope.getVariable("wasNull"));
                             block.append(scope.getVariable("wasNull").set(constantFalse()));
                             stackTypes.add(boolean.class);
                             currentParameterIndex++;
                             break;
                         case USE_BOXED_TYPE:
+                            block.append(arguments.get(realParameterIndex));
                             block.append(boxPrimitiveIfNecessary(scope, type));
                             block.append(scope.getVariable("wasNull").set(constantFalse()));
                             break;
+                        case BLOCK_POSITION:
+                            InputReferenceCompiler.InputReferenceNode blockPositionNode = (InputReferenceCompiler.InputReferenceNode) arguments.get(realParameterIndex);
```

Comment:
static import `InputReferenceNode`; Maybe call the variable `inputReferneceNode` ?